contrib/kafka/filters/network/source/mesh/BUILD

```diff
@@ -252,6 +252,7 @@ envoy_cc_library(
     ],
     tags = ["skip_on_windows"],
     deps = [
+        "//envoy/common:base_includes",
```

Comment:
Why do we even have this target still if folks don't need it? I think it was added because these seemed table stakes for any file, but it seems we don't need that anymore, e.g. `pure.h` makes its way in some other way? A bit confused on that dependency chain actually.

Reply:
Ok, I will split the base_includes into individual headers and fix broken dependencies in the follow up PR, as I expect it to be messier than this one.

I'm also removing implicit dep on the `absl/strings` in the next PR as it was causing issue in the internal build as well. So this clean-up is not finished yet.